tests/virt_autotest/sev_snp_guest_verification.pm

```diff
@@ -0,0 +1,1270 @@
+            $ret = execute_over_ssh(
+                address => $args{dst_machine},
+                command => "rpm -q $pkg",
+                assert => 0
+            );
+        }
+
+        # If package is installed, return true immediately
+        if ($ret == 0) {
+            record_info("Package found", "Package $pkg is installed on $location", result => 'ok');
+            return 1;
+        }
+    }
+
+    # If we get here, none of the packages were installed
+    record_info('No packages found', "None of the required packages are installed on $location", result => 'fail');
+    return 0;
+}
+
+=head2 install_snp_packages_on_guest
+
+  install_snp_packages_on_guest($self, guest_name => 'name', packages => ['pkg1', 'pkg2'])
+
+Install the necessary SEV-SNP packages on a guest machine.
+This function uses SSH to install packages on the remote guest.
+Returns 1 for success, 0 for failure.
+
+=cut
+
+sub install_snp_packages_on_guest {
+    my ($self, %args) = @_;
+    $args{guest_name} //= '';
+    $args{packages} //= [];
+
+    die 'Guest name must be provided for remote package installation' if ($args{guest_name} eq '');
+    die 'Package list must be provided for installation' if (!@{$args{packages}});
+
+    my $guest_name = $args{guest_name};
+    my $package_list = join(' ', @{$args{packages}});
+
+    record_info("Installing Packages", "Installing packages on guest $guest_name: $package_list");
+
+    # Use execute_over_ssh with better error handling and longer timeout for package installation
+    my $install_result = execute_over_ssh(
+        address => $guest_name,
+        command => "zypper --non-interactive in $package_list",
+        assert => 0,
+        timeout => 180    # Increased timeout for package installation
+    );
+    save_screenshot;
+
+    if ($install_result != 0) {
+        record_info('Installation Failed', "Failed to install packages on guest $guest_name: $package_list", result => 'softfail');
+        return 0;
+    }
+
+    record_info('Installation Attempted', "Attempted to install all required packages on guest $guest_name");
+    return 1;
+}
+
+##############################################
+#  SECTION 5: ATTESTATION VERIFICATION FUNCTIONS #
+##############################################
+
+=head2 verify_guest_attestation
+
+  verify_guest_attestation($self, guest_name => 'name')
+
+Generate and verify an attestation report for an SEV-SNP guest.
+
+=cut
+
+sub verify_guest_attestation {
+    my ($self, %args) = @_;
+    $args{guest_name} //= '';
+
+    die 'Guest name must be given to perform attestation verification.' if ($args{guest_name} eq '');
+
+    my $guest_name = $args{guest_name};
+
+    record_info('Guest Attestation', "Verifying guest attestation for $guest_name");
+
+    # Make sure guest is online
+    virt_autotest::utils::wait_guest_online($guest_name, 50, 1);
+
+    # Create a temporary directory for attestation artifacts
+    my $temp_dir = "/tmp/sev_snp_attestation_" . time();
```

Comment:
Similar here for log, putting is under TOP DIR defined in CONST?

Reply:
Done.